### PR TITLE
HTTPS (almost) all the docs

### PR DESCRIPTION
**Pull request by chrisfinazzo**

### CODE REVIEW

File: site/_docs/contributing.md [hunk 1]
```diff
@@ -27,9 +27,11 @@ following in mind:
   change is to review, the more likely it will be merged.
 * When submitting a pull request, please make judicious use of the pull request
   body. A description of what changes were made, the motivations behind the
-  changes and [any tasks completed or left to complete](http://git.io/gfm-tasks)
+  changes and [any tasks completed or left to complete][]
```
Comment (parkr) [inline site/_docs/contributing.md]: why not do this inline?

File: site/_docs/github-pages.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ branch]({{ site.repository }}/tree/gh-pages) of the same repository.
 <div class="note warning">
   <h5>Source Files Must be in the Root Directory</h5>
   <p>
-GitHub Pages <a href="https://help.github.com/articles/troubleshooting-github-pages-build-failures#source-setting">overrides</a> the <a href="http://jekyllrb.com/docs/configuration/#global-configuration">“Site Source”</a> configuration value, so if you locate your files anywhere other than the root directory, your site may not build correctly.
+GitHub Pages <a href="https://help.github.com/articles/troubleshooting-github-pages-build-failures#source-setting">overrides</a> the <a href="https://jekyllrb.com/docs/configuration/#global-configuration">“Site Source”</a> configuration value, so if you locate your files anywhere other than the root directory, your site may not build correctly.
```
Comment (parkr) [inline site/_docs/github-pages.md]: Let's remove `https://jekyllrb.com` here and just make it `/docs/configuration/#global-configuration`.

File: site/_docs/upgrading/2-to-3.md [hunk 3]
```diff
@@ -92,7 +92,7 @@ error when trying to **serve** or **build**:
 {% highlight text %}
 Since v3.0, permalinks for pages in subfolders must be relative to the site
 source directory, not the parent directory. Check
-http://jekyllrb.com/docs/upgrading/ for more info.
+https://jekyllrb.com/docs/upgrading/ for more info.
```
Comment (parkr) [inline site/_docs/upgrading/2-to-3.md]: This isn't true to the error message, please revert this one :)

**Comment by parkr** [comment]
Thanks!

File: site/_docs/contributing.md (hunk 1 above)
Comment (chrisfinazzo) [inline site/_docs/contributing.md]: If it's inline, the length is over 80 characters. While not a hard-and-fast rule, would this get caught by TravisCI?

File: site/_docs/github-pages.md (hunk 2 above)
Comment (chrisfinazzo) [inline site/_docs/github-pages.md]: OK,  will update.

File: site/_docs/contributing.md (hunk 1 above)
Comment (parkr) [inline site/_docs/contributing.md]: I think it's fine for this to be > 80 characters. Our code's limit is more like 110 or 120 characters. Thanks!

**Comment by parkr** [comment]
:)

**Pull request merged**